module3-crud-and-data-models/r1.1-principles-of-setting-up-your-schema/README.md

```diff
@@ -0,0 +1,58 @@
+
+Poorly designed databases can cause many problems, including wasting resources, making maintenance difficult, and hindering performance. That's why having a great database schema design is a crucial part of effective data management.
+By defining categories of data and relationships between those categories, database schema design makes data much easier to retrieve, consume, manipulate, and interpret.
+Without a clean, efficient, consistent database schema, you’ll struggle to make the best use of your enterprise data. For example, the same data might be duplicated in multiple locations—or even worse, might be inconsistent between these locations.
+
+### Type of database
+
+Most developers don't see the difference between relational database schema and MongoDB schema, but in reality, it is not the same
+
+<img src="https://i.imgur.com/3yobiPB.jpg" alt="meme" width="400"/>
+
+In a nutshell,
+in relational databases, models are usually independent of queries, there is no duplication of data as data will _mostly_ be separated into different tables, and it is rigid, you will have to define types and fields for your schema beforehand.
+
+in the other hand, `MongoDB`, you have more flexibility, there are no rules, documents can have new data or fields at any point of time, no need to define types.
+
+### Embedding vs. Referencing
+
+One of the key points to establish a good schema design (especially, if you are using MongoDB) is whether to embed or reference your data.
+
+#### When to `Embed`:
+
+- To retrive all data in a single query.
+- Avoid expense JOINs or $lookups.
+- Update all data with a single operation.
+- In one-to-one or one-to-many relationships between documents.
+
+#### When to `Reference`
+
+- If you need smaller documents or tables, your data will be separated across multiple documents or tables.
+- No duplicate of data.
+- To reduce accessed data not accessed on every query.
+- In many-to-many relationships between tables and documents.
+
+In general, there are few `rules` you can follow to better design your schema:
+
+##### 1. Favor embedding over referencing unless there is a compelling reason not to.
+
+Embedded documents are an efficient and clean way to store related data, especially data that's regularly accessed together. The more often a given workload can retrieve a single document and have all the data it needs, the more consistently high-performance your application will be.
+
+##### 2. Needing to access an object on its own is a compelling reason not to embed.
+
+Separate data that can be referred to from multiple places into its own collection.
+This is not so much a "storage space" issue as it is a "data consistency" issue. If many records will refer to the same data it is more efficient and less error prone to update a single record and keep references to it in other places.
+
+##### 3. Avoid JOINs and $lookups if they can be avoided, but don't be afraid if they can provide a better schema design.
```

Comment:
an example of embedding vs referencing with best practices is needed. It is well worded, but I feel it favors embedding, which might be not the case most of the time. 

We can use an example of _Student_ and _Classroom_, it can be done in embedding, but usually, a bad practice to embed Classroom in Student, or vice-versa. Another case is _User_ and _Interests_, based on the app use case, which can be embedded mostly if you won't be querying Interests specifically.

Reply:
I didn't go there as there will be a lot of talking and maybe needing visuals too, I added in the resources an article that goes in-depth talking about that, and I still don't think we should provide examples as it will still depend on the use case at the end, what do you think?